shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/datasource/props/DataSourceProperties.java

```diff
@@ -119,4 +122,84 @@ public int hashCode() {
         }
         return Objects.hashCode(dataSourceClassName, stringBuilder.toString());
     }
+
+    /**
+     * can data source be aggregate.
+     *
+     * @param dataSourceProperties data source properties
+     * @throws InvalidDataSourcePropertiesException invalid data source properties exception
+     */
+    public void dataSourceCanBeAggregate(final DataSourceProperties dataSourceProperties) throws InvalidDataSourcePropertiesException {
```

Comment:
How about change method name to `checkToBeAggregatedDataSources` ?

Reply:
Fix in https://github.com/apache/shardingsphere/pull/16632